engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1253,10 +1254,14 @@ public void run()
     private final Container container;
     private boolean stopRequested;
     private Token<StramDelegationTokenIdentifier> delegationToken;
+    private final NodeId nodeId;
+    private final String host;
 
     private AllocatedContainer(Container c)
     {
       container = c;
+      nodeId = c.getNodeId();
+      host = nodeId.getHost();
```

Comment:
Can getNodeId() return null here as well? Looking at Hadoop Container (and NodeId) Impl classes there are 2 variations: proto and builder. In the proto mode, nodeId etc are all set. If the same container object is used to send a request to Yarn and if any setter is called then the object switches to builder mode in which case nodeId will be null. So this bug might indicate possibly incorrect usage of objects in the alloatedContainers map.